sky/skylet/constants.py

```diff
@@ -152,10 +149,8 @@
     f'conda create -y -n {SKY_REMOTE_PYTHON_ENV_NAME} python=3.10 && '
     f'conda activate {SKY_REMOTE_PYTHON_ENV_NAME};'
     # Create a separate conda environment for SkyPilot dependencies.
-    # We use --system-site-packages to reuse the system site packages to avoid
-    # the overhead of installing the same packages in the new environment.
     f'[ -d {SKY_REMOTE_PYTHON_ENV} ] || '
-    f'{SKY_PYTHON_CMD} -m venv {SKY_REMOTE_PYTHON_ENV} --system-site-packages;'
+    f'{SKY_PYTHON_CMD} -m venv {SKY_REMOTE_PYTHON_ENV};'
```

Comment:
Can we add a comment here saying why we don't use the flag? Something like: `We don't use --system-site-packages here, though it can reuse the package in the base env, because it is unreliable when user upgrade the package in the base env. Reference: https://github.com/skypilot-org/skypilot/issues/4097`